plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java

```diff
@@ -756,16 +775,16 @@ public Object[][] clickHouseDateTimeMinMaxValuesDataProvider()
     @Test(dataProvider = "unsupportedTimestampDataProvider")
     public void testUnsupportedTimestamp(String unsupportedTimestamp)
     {
-        String minSupportedTimestamp = (String) clickHouseDateTimeMinMaxValuesDataProvider()[0][0];
-        String maxSupportedTimestamp = (String) clickHouseDateTimeMinMaxValuesDataProvider()[1][0];
+        String minSupportedTimestamp = (String) clickHouseDateTime64MinMaxValuesDataProvider()[0][1];
+        String maxSupportedTimestamp = (String) clickHouseDateTime64MinMaxValuesDataProvider()[1][1];
```

Comment:
You can instead have a single data provider which does `DataProviders.cartesianProduct`

Reply:
The reason I use a separate `dataProvider` is that different ClickHouse versions may support different ranges for DateTime or DateTime64, and we can provide different `dataProvider` by overriding them in different ClickHouse version tests.